onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -658,16 +658,141 @@ def aten_huber_loss_backward(
     raise NotImplementedError()
 
 
+def _get_im2col_indices_along_dim(
+    input_d: int,
+    kernel_size_d: int,
+    dilation_d: int,
+    padding_d: int,
+    stride_d: int,
+):
```

Comment:
I see all the inputs here are expected to be known at export-time? That is, this is in trace-only mode? Why do we need to encode this logic using ONNX operators, instead of doing it directly in Python? (The ONNX encoding is useful if and when the operator becomes more dynamic, with some of these parameters being runtime-values. However, as it exists, this will become dependent on subsequent constant-folding for the current static case.)

Reply:
Good point. At least input_d should be dynamic I think?

Reply:
Ok. I guess it is just a matter of changing the type signature above for input_dy